### PR TITLE
QUARKS-232 Change name from Quarks to Edgent

### PR DESCRIPTION
**Pull request by dlaboss**
Rename everything:
- packages, name occurrences in javadoc, build scripts, sample scripts, *.md

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -125,8 +125,8 @@ A couple of key items in the mean time:
 Search the code for org.slf4j.LoggerFactory to see a sample of its use.
 
 ### Use of Java 8 features
-Quarks primary development environment is Java 8, to take advantage of lambda expressions
-since Quarks' primary api is a functional one.
+Edgent primary development environment is Java 8, to take advantage of lambda expressions
+since Edgent' primary api is a functional one.
```
Comment (queeniema) [inline DEVELOPMENT.md]: nit: _Quarks'_ should be _Edgent's_

Comment (dlaboss) [inline DEVELOPMENT.md]: good eyes! thx.  I checked and this was the only instance of that transformation issue.

**Comment by dlaboss** [comment]
Noticed there was an old quarks-edge reference which is now edgent-edge.
What's the appropiate value for this manifest attr?

                <attribute name="Implementation-Vendor" value="edgent-edge@github.com"/>

File: android/hardware/src/main/java/quarks/android/hardware/SensorStreams.java [hunk 2]
```diff
@@ -16,14 +16,14 @@ Licensed to the Apache Software Foundation (ASF) under one
 specific language governing permissions and limitations
 under the License.
 */
-package quarks.android.hardware;
+package edgent.android.hardware;
```
Comment (Cazen) [inline android/hardware/src/main/java/quarks/android/hardware/SensorStreams.java]: Android related directory does not seems that changed to edgent

- android/topology/src/main/java/quarks/android/oplet/RunOnUIThread.java
- android/hardware/src/main/java/quarks/android/hardware/SensorStreams.java
- android/topology/src/main/java/quarks/android/topology/ActivityStreams.java
- android/hardware/src/main/java/quarks/android/hardware/runtime/SensorSourceSetup.java
- android/hardware/src/main/java/quarks/android/hardware/runtime/SensorChangeEvents.java

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -126,7 +126,7 @@ Search the code for org.slf4j.LoggerFactory to see a sample of its use.
 
 ### Use of Java 8 features
 Edgent primary development environment is Java 8, to take advantage of lambda expressions
```
Comment (queeniema) [inline DEVELOPMENT.md]: nit: Didn't notice this the first time, but there was a typo originally (missing apostrophe). _Edgent_ should be _Edgent's_.

File: android/hardware/src/main/java/quarks/android/hardware/SensorStreams.java (hunk 2 above)
Comment (dlaboss) [inline android/hardware/src/main/java/quarks/android/hardware/SensorStreams.java]: Thanks!  I see there are a couple of renames that I missed:
./android/hardware/src/main/java/quarks/...
./android/topology/src/main/java/quarks/...
./console/servlets/webapp_content/js/ext/sankey_quarks.js
./console/servlets/webapp_content/resources/images/small_quarks.png
./scripts/runhelloquarks.cmd
./scripts/runhelloquarks.sh

**Comment by ddebrunner** [comment]
-1

Top level package should be `org.apache.edgent`, not just `edgent`.

We had postponed moving to org.apache until the name was settled, now the name is assumed to be settled we should have a single change to `org.apache.edgent`.

**Comment by dlaboss** [comment]
now you tell me :-)

**Comment by ddebrunner** [comment]
@dlaboss Sorry, thought it was well known that the package names would be `org.apache.something` ...

**Comment by dlaboss** [comment]
@ddebrunner no worries. I just forgot.  Unless I hear otherwise, I'm leaving the jars with the simple "edgent." prefix (e.g., edgent.providers.direct.jar).

**Comment by ddebrunner** [comment]
@dlaboss For the jar names I would follow what most Apache projects do. Haven't looked yet, but I seem to remember org.apache in the names

**Comment by ddebrunner** [comment]
My veto is lifted since the package names are now org.apache.*

**Comment by dlaboss** [comment]
A quick sampling of other projects (Hadoop, Hive, Derby, Kafka) shows an absence of org.apache in jar names.  

That said, http://incubator.apache.org/guides/release-java.html#best-practice-jars has a "TODO" that seems to indicate apache should be in the name.

So do we want "apache.edgent" as the prefix for our jar names?  Need a decision ASAP.

**Comment by ddebrunner** [comment]
@dlaboss Yeah, I'd thought it was common to have org.apache in the jar names, but the majority I see are also without that. I think I was remembering org.eclipse jars.

**Pull request merged**